File: C/Factors-Of-A-Number.c

// Question : Given a number find all the factors of that number.
//          Input : User input a number
//          Output : Factors of that number

#include <stdio.h>
int main() 
{
    int i,num;
    printf("\nEnter a Number to find factors : ");
    scanf("%d", &num);
    printf("Factors of %d are: ", num);
    for (i = 1; i <= num; ++i) 
    {
        if (num % i == 0) 
        {
            printf("%d ", i);
        }
    }

   printf("\n");
   return 0;
}

File: Python/Factors-Of-A-Number.py

# Question : Given a number find all the factors of that number.
#          Input : User input a number
#          Output : A list of factors of that number

num=int(input("Enter a number to find factors : "))
l=[]

def factors(x):
    print("The factors of",x,"are :")
    for i in range(1,x+1):
        if x % i == 0:
            l.append(i)
    print(l) 

factors(num)
